python/ribasim/ribasim/model.py

```diff
@@ -342,3 +347,58 @@ def plot(self, ax=None, indicate_subnetworks: bool = True) -> Any:
         ax.legend(handles, labels, loc="lower left", bbox_to_anchor=(1, 0.5))
 
         return ax
+
+    def to_xugrid(self) -> xugrid.UgridDataset:
+        """Convert the network to a xugrid.UgridDataset."""
+        node_df = self.node_table().df
```

Comment:
I think that method should throw an Exception early if `xugrid` is `None`

Reply:
I use this in imod-python or xugrid:

```python
class MissingOptionalModule:
    """
    Presents a clear error for optional modules.
    """

    def __init__(self, name):
        self.name = name

    def __getattr__(self, name):
        raise ImportError(f"{self.name} is required for this functionality")
```

But here probably `if xugrid is None` suffices.

Reply:
Using `MissingOptionalModule` sounds better to me.
Less chance to overlook the assertion, since mypy doesn't seem to spot it.

Reply:
Done in d822065d3f1708940e973a7059d7075eaa6537de.